PocketKit/Sources/Sync/RemoteMapping/SavedItem+RemoteMapping.swift

```diff
@@ -24,17 +24,31 @@ extension SavedItem {
     }
 
     public func update(from remote: RemoteSavedItem, with space: Space) {
+        guard let context = managedObjectContext else {
+            Log.capture(message: "Managed context was nil")
+            return
+        }
+
         remoteID = remote.remoteID
         url = remote.url
         createdAt = Date(timeIntervalSince1970: TimeInterval(remote._createdAt))
         deletedAt = remote._deletedAt.flatMap(TimeInterval.init).flatMap(Date.init(timeIntervalSince1970:))
         archivedAt = remote.archivedAt.flatMap(TimeInterval.init).flatMap(Date.init(timeIntervalSince1970:))
         isArchived = remote.isArchived
         isFavorite = remote.isFavorite
-
-        guard let context = managedObjectContext else {
-            Log.capture(message: "Managed context was nil")
-            return
+        if let annotations = remote.annotations, let remoteHighlights = annotations.highlights, !remoteHighlights.isEmpty {
```

Comment:
What if a user had highlights, but then deletes them from the web? I think this skips that case? Also if thats followup and the main thing here was highlight blob creation ignore me!

Reply:
good catch @bassrock ! Indeed if no highlights are found, we should clear them